urllib3/connectionpool.py

```diff
@@ -383,8 +383,13 @@ def _make_request(self, conn, method, url, timeout=_Default, chunked=False,
         try:
             try:  # Python 2.7, use buffering of HTTP responses
                 httplib_response = conn.getresponse(buffering=True)
-            except TypeError:  # Python 2.6 and older
-                httplib_response = conn.getresponse()
+            except TypeError:  # Python 2.6 and older, Python 3
+                try:
+                    httplib_response = conn.getresponse()
+                except Exception as e:
```

Comment:
This is rather overbroad. It would be substantially better if we caught *and verified* that this exception is the given `TypeError` before hiding this part of the traceback.